components/sup/src/manager/self_updater.rs

```diff
@@ -49,48 +46,24 @@ impl SelfUpdater {
             update_channel: ChannelIdent)
             -> Receiver<PackageInstall> {
         let (tx, rx) = oneshot::channel();
-        // Execute this future on a dedicated thread, and using a
-        // single-threaded runtime. Eventually, this should use `tokio::spawn`,
-        // but that will require refactoring to make the future safe
-        // to spawn on an executor.
-        //
-        // (Note: the `enable_all` is to ensure the Tokio timer is set
-        // up on the runtime, which is needed both here and also in
-        // the async reqwest calls we make from here.)
-        thread::Builder::new().name("self-updater".to_string())
-                              .spawn(move || {
-                                  let mut rt =
-                                      runtime::Builder::new().basic_scheduler()
-                                                             .enable_all()
-                                                             .build()
-                                                             .expect("Could not spawn runtime \
-                                                                      for self-updater thread!");
-
-                                  rt.block_on(Self::run(tx, &current, &update_url, &update_channel))
-                              })
-                              .expect("Unable to start self-updater thread");
+        tokio::spawn(Self::run(tx, current, update_url, update_channel));
         rx
     }
 
     async fn run(tx: Sender<PackageInstall>,
-                 current: &PackageIdent,
-                 update_url: &str,
-                 update_channel: &ChannelIdent) {
+                 current: PackageIdent,
+                 update_url: String,
+                 update_channel: ChannelIdent) {
         debug!("Self updater current package, {}", current);
         // SUP_PKG_IDENT will always parse as a valid PackageIdent,
         // and thus a valid InstallSource
         let install_source: InstallSource = SUP_PKG_IDENT.parse().unwrap();
         loop {
             let next_check = SteadyTime::now() + TimeDuration::milliseconds(update_frequency());
 
-            match util::pkg::install(// We don't want anything in here to print
-                                     &mut UI::with_sinks(),
-                                     &update_url,
-                                     &install_source,
-                                     &update_channel).await
-            {
+            match util::pkg::install_no_ui(&update_url, &install_source, &update_channel).await {
                 Ok(package) => {
-                    if current < package.ident() {
+                    if &current < package.ident() {
                         debug!("Self updater installing newer Supervisor, {}",
                                package.ident());
                         tx.send(package).expect("Main thread has gone away!");
```

Comment:
> It may also help clean up some of the time calculation code.

Related to this, do you think we need to precisely calculate how long to wait between installs? Opposed to simply having a constant delay.

Reply:
Addressed in [4da281](https://github.com/habitat-sh/habitat/pull/7455/commits/4da281925a64de711d6150e9d3e10f6591cfa168)

Reply:
Yeah, the constant delay is fine (in addition to being simpler! :sweat_smile:). I think all the delay calculation is a vestige of the pre-futures way of handling all this. I think any time we see stuff like that, we can simplify it like you did in https://github.com/habitat-sh/habitat/pull/7455/commits/4da281925a64de711d6150e9d3e10f6591cfa168.

Nice!